chia/rpc/wallet_rpc_api.py

```diff
@@ -637,17 +637,23 @@ async def create_new_wallet(self, request: Dict) -> EndpointResult:
             # If not provided, the name will be autogenerated based on the tail hash.
             name = request.get("name", None)
             if request["mode"] == "new":
-                async with self.service.wallet_state_manager.lock:
-                    cat_wallet: CATWallet = await CATWallet.create_new_cat_wallet(
-                        wallet_state_manager,
-                        main_wallet,
-                        {"identifier": "genesis_by_id"},
-                        uint64(request["amount"]),
-                        name,
+                if request["test"]:
```

Comment:
Maybe make this optional? `request.get("test", False)`

Reply:
Ah yeah good call out (fixed)